src/levelui.cpp

```diff
@@ -296,6 +302,7 @@ void Level::RenderUI()
 	{
 		if (ImGui::Begin("Checkpoints", &w_checkpoints, ImGuiWindowFlags_AlwaysAutoResize))
 		{
+			ImGui::InputTextWithHint("Search##", "Search Quadblocks...", QUADBLOCK_SEARCH_BUFFER, IM_ARRAYSIZE(QUADBLOCK_SEARCH_BUFFER));
```

Comment:
Prefer using the C++ ImGui API for string functions. These are defined in `imgui/misc/cpp/imgui_stdlib.h`.

```
namespace ImGui
{
    // ImGui::InputText() with std::string
    // Because text input needs dynamic resizing, we need to setup a callback to grow the capacity
    IMGUI_API bool  InputText(const char* label, std::string* str, ImGuiInputTextFlags flags = 0, ImGuiInputTextCallback callback = nullptr, void* user_data = nullptr);
    IMGUI_API bool  InputTextMultiline(const char* label, std::string* str, const ImVec2& size = ImVec2(0, 0), ImGuiInputTextFlags flags = 0, ImGuiInputTextCallback callback = nullptr, void* user_data = nullptr);
    IMGUI_API bool  InputTextWithHint(const char* label, const char* hint, std::string* str, ImGuiInputTextFlags flags = 0, ImGuiInputTextCallback callback = nullptr, void* user_data = nullptr);
}
```